File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    let answer;
    let [f0, f1] = [0 , 1];
    let i = 2;
    
    while (i !== n) {
        [f0, f1] = [f1 % 1234567 , (f0 + f1) % 1234567];
        i++
    }
    answer = f0 + f1;
    return answer % 1234567;
}

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    result = [0 for _ in range(n + 1)]
    result[0] = 0
    result[1] = 1
    i = 2
    
    while not result[n]:
        result[i] = result[i - 2] + result[i - 1]
        i += 1
    
    return result[n] % 1234567